metabase/metabase
# Bulk Filter Modal: Inline Boolean Filter Input

## Description

- Instead of using the same dropdown + popover interface for all filter types, we want to move most of them inline so they can be filtered in fewer clicks
- This creates a new parent component `<InlineFilterSelect />` that will be the parent for all of the different field types, but currently only supports booleans
- Creates checkbox variant of the boolean picker with shared logic from he radio version
- **null/not null**: currently any existing null/not null filters on a boolean field will set both checkboxes to indeterminate state
![Screen Shot 2022-05-27 at 3 57 58 PM](https://user-images.githubusercontent.com/30528226/170794196-6f523868-ec2a-4fd3-a574-9c89595cb8d2.png)

## Screenshots

![Screen Shot 2022-05-27 at 3 50 04 PM](https://user-images.githubusercontent.com/30528226/170794041-1284c84d-f62b-45e8-8b9b-9a8aa284523f.png)


## Testing Steps

- open a table with boolean data
- open the bulk filter modal and set true/false filters on a boolean column
- see that the filters save
- add/edit boolean filters in notebook mode and see that they populate properly when opening the bulk filter modal
- add a null/not null filter on a boolean column in notebook mode
- see that it displays indeterminate checkboxes in the bulk filter modal
- in the bulk filter modal, uncheck all boolean checkboxes, see that it removes the boolean filter altogether
